proto/v1/api/android.proto

```diff
@@ -0,0 +1,391 @@
+  // operation. This is only a request, and may not be honoured by the server (which will only
+  // adhere to this field on a best-effort basis). This is calculated by taking the wire byte size
+  // (without compression) corresponding to a computed response (though the server may use estimates
+  // for efficiency, and will guarantee that the estimate isn't far off from the actual size sent).
+  uint32 requested_max_payload_size_bytes = 4;
+}
+
+// The expected response from the server when provided with a TopicContentRequest.
+message TopicContentResponse {
+  option (org.oppia.proto.v1.versions.api_proto_version_type) = TOPIC_CONTENT_REQUEST_RESPONSE_PROTO_VERSION;
+
+  // The common proto version defining this request/response pair.
+  org.oppia.proto.v1.versions.TopicContentRequestResponseProtoVersion proto_version = 1;
+
+  // A list of the individual items that were downloaded. Items in this list are expected to be
+  // relatively in the same order as the identifiers in the request, but some items may be missing
+  // (see the request's documentation for details). Note also that these will never include external
+  // assets such as voiceovers or images--those always must be downloaded separately.
+  repeated DownloadResult download_results = 2;
+
+  // The result of downloading a specific structure. This may either contain the result of the
+  // download, or an indication that the download was skipped and roughly why. The potential results
+  // correspond to the download requests that can be made per DownloadRequestStructureIdentifier.
+  message DownloadResult {
+    option (org.oppia.proto.v1.versions.api_proto_version_type) = TOPIC_CONTENT_REQUEST_RESPONSE_PROTO_VERSION;
+
+    // The entity type and ID corresponding to this download. This is guaranteed to exactly match at
+    // least one of the original identifiers in the request.
+    DownloadRequestStructureIdentifier identifier = 1;
+
+    // The type/nature of the download result (see fields for specifics).
+    oneof result_type {
+      // The presence of this value indicates that the request corresponding to this particular
+      // identifier was skipped and suggested to retry by the Oppia backend. Note that the presence
+      // of this value in the oneof is sufficient to indicate its state; its actual value does not
+      // matter.
+      bool skipped_should_retry = 2;
+
+      // The presence of this value indicates that the request outright failed for this particular
+      // identifier, potentially due to incorrect version/identifier information or an internal
+      // server failure. It is not recommended to retry the request since it's expected to fail
+      // again. Note that the presence of this value in the oneof is sufficient to indicate its
+      // state; its actual value does not matter. Note also that this is different than an HTTP
+      // error since certain error codes can be retried (in those cases, the entire response is
```

Comment:
Don't really understand this last sentence -- the first part of the docstring says "the request outright failed" and here the parenthetical is saying "the entire response is expected to fail", so I don't see what the difference is here.